### DISCUSSION TITLE
请问如何在UI里让一定范围内的按钮和插槽之类的产生联动？

### DISCUSSION THREAD

**Discussion by cimdy** [mouse0w0/MinecraftDeveloperGuide#89] [问答 / Q&A]
如图，假设我创建了一个天赋树，因为天赋树我也不会写，假设，然后在插槽中镶嵌了一个珠宝，该珠宝会在自身为中心的位置，或者某个其他天赋点的位置渲染一个某种大小的圆环，使得圆环范围内的其他天赋不用链接就可以被点出，或者不用点出就可以直接生效，或者被替换为其他属性，大概需要什么思路去实现它。
![OIP-C](https://github.com/user-attachments/assets/910df2d4-d330-44c5-b02f-02f7dfd1acf8)

**Comment by mouse0w0** [comment]
做一遍碰撞检测就行，遍历所有天赋的圆心是否在对应环的外圆，并而不在对应环的内圆。这个碰撞检测可以缓存起来。

**Reply by cimdy** [reply]
待我学成归来
